tests/packages/test_locker.py

```diff
@@ -753,3 +754,77 @@ def test_content_hash_with_legacy_is_compatible(
     content_hash = locker._get_content_hash()
 
     assert (content_hash == old_content_hash) or fresh
+
+
+def test_lock_file_resolves_file_url_symlinks(root: ProjectPackage):
+    """
+    Create directories and file structure as follows:
+
+    d1/
+    d1/testsymlink -> d1/d2/d3
+    d1/d2/d3/lock_file
+    d1/d4/source_file
+
+    Using the testsymlink as the Locker.lock file path should correctly resolve to
+    the real physical path of the source_file when calculating the relative path
+    from the lock_file, i.e. "../../d4/source_file" instead of the unresolved path
+    from the symlink itself which would have been "../d4/source_file"
+
+    See https://github.com/python-poetry/poetry/issues/5849
+    """
+    with tempfile.TemporaryDirectory() as d1:
+        symlink_path = Path(d1).joinpath("testsymlink")
+        with tempfile.TemporaryDirectory(dir=d1) as d2, tempfile.TemporaryDirectory(
+            dir=d1
+        ) as d4, tempfile.TemporaryDirectory(dir=d2) as d3, tempfile.NamedTemporaryFile(
+            dir=d4
+        ) as source_file, tempfile.NamedTemporaryFile(
+            dir=d3
+        ) as lock_file:
+            lock_file.close()
+            os.symlink(Path(d3), symlink_path)
```

Comment:
```suggestion
            try:
                os.symlink(Path(d3), symlink_path)
            except OSError:
                pass
                if sys.platform == "win32":
                    # os.symlink requires either administrative privileges or developer
                    # mode on Win10, throwing an OSError if neither is active.
                    # Test is not possible in that case.
                    return
                raise
```

`os.symlink` requires os.symlink requires either administrative privileges or developer mode on Win10. We have a workaround to avoid symlinks in https://github.com/python-poetry/poetry/blob/83e6c318f59a77e83553c9805562490a4e5f8811/tests/helpers.py#L75-L92

However, I don't think it makes sense to use this function here. Thus, maybe it's best to "skip" the test for windows when `os.symlink` fails.

Reply:
Done